### PR TITLE
feat(backend): update models and controllers to async/await

### PR DESCRIPTION
**Pull request by pakyardim**
This commit migrates from "mysql" to "mysql2" package to update models/controllers to async/await while also increasing the security level of the backend.

### CODE REVIEW

File: server/controllers/user.js [hunk 1]
```diff
@@ -8,98 +8,81 @@ exports.create = async (req, res) => {
     return res.status(400).send({ message: 'Content can not be empty!' });
   }
 
-  User.findByEmail(req.body.email, (err, data) => {
-    if (data) {
+  try {
+    const emailExist = await User.findByEmail(req.body.email);
+    console.log(emailExist);
```
Comment (kaganecee) [inline server/controllers/user.js:13]: There is a missing log here.

File: server/models/period.js [hunk 2]
```diff
@@ -7,126 +7,126 @@ const Period = function (period) {
...
-        return result({ kind: 'not_found' }, null);
-      }
-      console.log(`updated period with user_id ${user_id} and date ${today}: `, {
-        ...updatedPeriod,
-        user_id,
-        today
-      });
-      result(null, { ...updatedPeriod, user_id, today });
+  try {
+    const [rows] = await pool.execute(
```
Comment (kaganecee) [inline server/models/period.js:100]: why we are declaring a variable inside brackets

**Review by kaganecee — APPROVED** [review]

File: server/controllers/user.js (hunk 1 above)
Comment (pakyardim) [reply server/controllers/user.js:13]: oops

File: server/models/period.js (hunk 2 above)
Comment (pakyardim) [reply server/models/period.js:100]: array destructuring assignment. this way we are extracting the first element of the returned array from "pool.execute" process, and assigning it to the variable "rows"

**Pull request merged**